packages/redis/src/lib/RedisProvider.ts

```diff
@@ -540,47 +540,80 @@ export class RedisProvider<StoredValue = unknown> extends JoshProvider<StoredVal
 
   public async [Method.Random](payload: Payload.Random<StoredValue>): Promise<Payload.Random<StoredValue>> {
     const docCount = (await this[Method.Size]({ method: Method.Size, errors: [] })).data || 0;
+    const { unique } = payload;
 
-    if (docCount === 0) return { ...payload, data: [] };
-    if (docCount < payload.count) {
+    if (unique && docCount < payload.count) {
       payload.errors.push(this.error({ identifier: CommonIdentifiers.InvalidCount, method: Method.Random }, { count: payload.count, docCount }));
 
       return payload;
     }
 
+    if (docCount === 0) {
+      payload.errors.push(this.error({ identifier: CommonIdentifiers.MissingData, method: Method.Random }, { count: payload.count, docCount }));
+
+      return payload;
+    }
+
     const keys = await this[Method.Keys]({ method: Method.Keys, errors: [] });
 
     keys.data = keys.data || [];
     payload.data = [];
 
-    for (let i = 0; i < payload.count; i++) {
-      const key = keys.data[Math.floor(Math.random() * keys.data.length)];
-      const getPayload = await this[Method.Get]({ method: Method.Get, errors: [], key, path: [] });
+    if (unique) {
+      const randomKeys = new Set<string>();
+
+      while (randomKeys.size < docCount) randomKeys.add(keys.data[Math.floor(Math.random() * keys.data.length)]);
```

Comment:
I'm very sure redis has a random way of getting documents

Reply:
https://redis.io/commands/srandmember/

Reply:
nvm this appears to be for a set stored in redis. There is randomKey but it just gets one so its probably just as simple to do this

Reply:
@dan-online can you do this, this is your provider after all.

Reply:
Yep of course